bigtable-dataflow-parent/bigtable-hbase-beam/src/test/java/com/google/cloud/bigtable/beam/CloudBigtableIOReaderTest.java

```diff
@@ -162,6 +204,230 @@ public void testSplits() throws IOException {
+    byte[] startKey = "A".getBytes();
+    byte[] endKey = "B".getBytes();
+
+    List<ReadRowsResponse> responses = generateResponses(fakeService, "A", "B", true, 10);
+
+    responses.remove(responses.size() - 1);
+
+    ReadRowsRequest request =
+        ReadRowsRequest.newBuilder()
+            .setTableName("projects/project/instances/instance/tables/table")
+            .setRows(
+                RowSet.newBuilder()
+                    .addRowRanges(
+                        RowRange.newBuilder()
+                            .setStartKeyClosed(ByteString.copyFrom(startKey))
+                            .setEndKeyOpen(ByteString.copyFrom(endKey))
+                            .build())
+                    .build())
+            .build();
+
+    CloudBigtableScanConfiguration config =
+        new CloudBigtableScanConfiguration.Builder()
+            .withProjectId("project")
+            .withInstanceId("instsance")
+            .withTableId("table")
+            .withRequest(request)
+            .withConfiguration(
+                BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY, "localhost:" + port)
+            .withConfiguration(BigtableOptionsFactory.BIGTABLE_TEST_IDLE_TIMEOUT_MS, "15000")
+            .build();
+
+    CloudBigtableIO.Source source = (CloudBigtableIO.Source) CloudBigtableIO.read(config);
+    CloudBigtableIO.Reader reader = (CloudBigtableIO.Reader) source.createReader(null);
+
+    List<Result> actual = new ArrayList<>();
+
+    // Reader.start() will read the first row
+    reader.start();
+    actual.add(reader.getCurrent());
+    int count = 1;
+
+    boolean sleep = false;
+    while (reader.advance()) {
+      count++;
+      actual.add(reader.getCurrent());
+      if (!sleep) {
+        Thread.sleep(20 * 1000);
+        sleep = true;
+      }
+    }
+
+    // Make sure idle timeout is retried
+    Assert.assertTrue(fakeService.count.get() > 1);
+    Assert.assertEquals(responses.size(), count);
+    Assert.assertEquals(
+        responses.stream()
+            .map(response -> response.getChunks(0).getRowKey())
+            .collect(Collectors.toList()),
+        actual.stream()
+            .map(result -> ByteString.copyFrom(result.getRow()))
+            .collect(Collectors.toList()));
+  }
+
+  @Test
+  public void testDisableRetryIdleTimeout() throws Exception {
+    byte[] startKey = "A".getBytes();
+    byte[] endKey = "B".getBytes();
+
+    List<ReadRowsResponse> responses = generateResponses(fakeService, "A", "B", true, 10);
+
+    responses.remove(responses.size() - 1);
+
+    ReadRowsRequest request =
+        ReadRowsRequest.newBuilder()
+            .setTableName("projects/project/instances/instance/tables/table")
+            .setRows(
+                RowSet.newBuilder()
+                    .addRowRanges(
+                        RowRange.newBuilder()
+                            .setStartKeyClosed(ByteString.copyFrom(startKey))
```

Comment:
I would use open start ranges so that your service can treat them the same as resumption requests